moveos/moveos-stdlib/rooch-framework/sources/events.move

```diff
@@ -0,0 +1,87 @@
+/// The Event module defines an `EventHandleGenerator` that is used to create
+/// `EventHandle`s with unique GUIDs. It contains a counter for the number
+/// of `EventHandle`s it generates. An `EventHandle` is used to count the number of
+/// events emitted to a handle and emit events to the event store.
+module rooch_framework::events {
+    use std::error;
+    use std::bcs;
+    use std::signer;
+    use std::vector;
+    use moveos_std::account_storage;
+    use moveos_std::storage_context::{StorageContext};
+
+    /// A resource representing the counter used to generate uniqueness under each account. There won't be destructor for
+    /// this resource to guarantee the uniqueness of the generated handle.
+    struct EventHandleGenerator has key {
+        // A monotonically increasing counter
+        counter: u64,
+        addr: address,
+    }
+
+    /// A handle for an event such that:
+    /// 1. Other modules can emit events to this handle.
+    /// 2. Storage can use this handle to prove the total number of events that happened in the past.
+    struct EventHandle<phantom T: drop + store> has store {
+        /// Total number of events emitted to this event stream.
+        counter: u64,
+        /// A globally unique ID for this event stream.
+        guid: vector<u8>,
```

Comment:
Use the ObjectID as `guid` type? like the TableHandle

Reply:
Maybe i can use the Vec[] corresponding to the ObjectID, and then can use the Move VM related interface.